### DISCUSSION TITLE
Troubleshooting lag in tracking with videos that have a drop in framerate

### DISCUSSION THREAD

**Discussion by ryoiwata** [talmolab/sleap#1115] [Help!] [answered]
- Hey Sleap Team! Thank you so much for developing such a wonderful tool to automate pose tracking!! We were having a little trouble with tracking videos that came from the Anymaze program([https://www.any-maze.com/](https://www.any-maze.com/)). The potential issue is that the skeleton for the predicted poses are lagging behind one second in the rendered video and predicting early by about one second in the `predicted.slp` file. In other words, the skeleton goes to where the body part was in the video either one second ago or one second later. The delay is consistent across the rendered video as well as the `predictions.slp` file. Fortunately, the shape and location of the predicted skeleton matches well with what the video was displaying a second ago or a second ahead.
- We are guessing that it’s an issue with the original video itself. With Anymaze, the frame rate seems to have been lowered to 24(from the originally set 30) for a few videos because we were recording from 4 cameras simultaneously. So the frame rate might have dropped in certain parts due to the high volume of data being recorded. We’ve noticed that when we do 20 second trial based clips(as opposed to the whole 30 minute video), there is no lag at all. Also, when we watch the original video there is a black border around the perimeter as if trying to correct for the resolution. (The video’s resolution is 1024x768)
- Here are the other specs according to Dropbox
    - Resolution
    1024x768
    Duration
    00:31:05
    Frame Rate
    24 fps
    Bit Rate
    2041 kb/s
    Codec
    h264
- Here is the link to the files that we used and generated. [https://www.dropbox.com/sh/sybji4zw1q6ctym/AADOD-EXZpVVSVqR5vgpr4P-a?dl=0](https://www.dropbox.com/sh/sybji4zw1q6ctym/AADOD-EXZpVVSVqR5vgpr4P-a?dl=0) Included are the sysout and syserr for the training as well as the model, configuration file, predicted pose, and labeled frames.
- For the training, we used 100 labeled frames from the video that was being predicted on only. And we created the Sleap Environment (version 1.2.9) with this command: `conda create -y -p ./sleap_1_2_9_env -c sleap -c nvidia -c conda-forge sleap=1.2.9`
- The command for training: `sleap-train /nancy/projects/sleap_pose_tracking/results/2022_12_21_lagging_tracking_troubleshoot/data/baseline_medium_rf.bottomup.json /nancy/projects/sleap_pose_tracking/results/2022_12_21_lagging_tracking_troubleshoot/data/10-03-22_Test_25_1-2v1-3.mp4.labeled.pkg.slp`
- The command for tracking: `sleap-track /nancy/projects/sleap_pose_tracking/results/2022_12_21_lagging_tracking_troubleshoot/data/10-03-22_Test_25_1-2v1-3.mp4 --tracking.tracker flow --tracking.similarity centroid --tracking.match greedy --tracking.clean_instance_count 2 --tracking.target_instance_count 2 -m /nancy/projects/sleap_pose_tracking/results/2022_12_21_lagging_tracking_troubleshoot/models/baseline_medium_rf.bottomup`
- Please let us know if you have any suggestions or if any other information would be useful. Thank you so much for continually being involved with the community and working on improving Sleap!
![Screenshot from 2022-12-21 18-47-01](https://user-images.githubusercontent.com/42358679/209024464-6ebc97f4-f1c0-462a-8c93-80caee71b9ba.png)
![Screenshot from 2022-12-21 18-46-17](https://user-images.githubusercontent.com/42358679/209024465-663a9475-ed5c-4630-bc84-b2e1cbce9858.png)

**Comment by roomrys** [comment, marked as answer]
Hi @ryoiwata,

The problem could be that the video is not [reliably seekable](http://blog.loopbio.com/video-io-1-introduction.html). Try [reencoding your video](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) and let us know how that works.

Thanks,
Liezl
